crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -966,6 +898,129 @@ impl Target {
 
 // Rust native methods
 impl Target {
+    /// Adds a [PackedOperation] to the [Target].
+    ///
+    /// Said addition results in a [NormalOperation] in the [Target] as variadics
+    /// are not yet supported natively. If no properties are specified the operation
+    /// is believed to be `Global` with properties `{Qargs::Global: None}`.
+    pub fn add_instruction<T>(
+        &mut self,
+        operation: PackedOperation,
+        params: &[Param],
+        name: Option<&str>,
+        props_map: Option<T>,
+    ) -> Result<(), TargetInvalidInstError>
+    where
+        T: Into<PropsMap>,
+    {
+        let name = if let Some(name) = name {
+            name.to_owned()
+        } else {
+            operation.name().to_string()
```

Comment:
Why `to_string()` here vs `to_owned()` above? Both branches you're converting an `&str` to a `String`.